compiler/match_/inference/type_seeder.rs

```diff
@@ -641,7 +641,7 @@ impl BinaryConstraint for Has<Variable> {
         owner
             .get_owns(seeder.snapshot, seeder.type_manager)?
             .iter()
-            .map(|(attribute, _)| TypeAnnotation::Attribute(attribute.clone()))
```

Comment:
There was a "bug": get_capabilities returned HashMap<InterfaceType, Capability>, where InterfaceType is always exactly the same as Capability::interface(). However, it may be worth returning Capability as a key and ObjectType (the source of this capability from inheritance) as value. We don't use it anywhere right now, so I didn't care, but I left a todo for it (don't want to spend additional time now).

Reply:
We still have a way to go from one capability to the original declarer easily right? Then we don't need this right now, i agree

Reply:
Yeah, I lagged when adding this comment tbh. We don't need a HashMap here because a capability already contains {source, interface_type}. So no more refactoring is needed